woocommerce/woocommerce
# k6: add additional merchant and API order tests

### All Submissions:






### Changes proposed in this Pull Request:



This PR adds and updates k6 requests for Merchant Order Flows and API Order Flows
-  Updates Search Orders to include more search types
- Adds additional Merchant Order requests (Closes https://github.com/woocommerce/woocommerce/issues/30239)
-- Add Order
-- Open Order
-- Update Order
- Adds API Order requests (Part of 250-gh-woocommerce/woocommerce-quality k6: Spike: API Testing) 
-- Create Order
-- Retrieve Order
-- List Orders
-- Update Order
-- Delete Order
-- Batch Create Order
-- Batch Update Order

### How to test the changes in this Pull Request:

1. Setup test environment
[See instructions here for more details
](https://github.com/woocommerce/woocommerce/tree/trunk/plugins/woocommerce/tests/performance#test-environment)
- use wp-env to setup a local environment with `performance/bin/init-sample-products.sh` that will set up a shop with sample products imported and the shop settings (payment method, permalinks, address etc) needed for the tests already set with the exception of coupon data.
- OR use an existing test site and update `performance/config.js` to use that site


2. Install k6 and run test
run `brew install k6`
[For other platforms please see the k6 installation guide.](https://k6.io/docs/getting-started/installation/)
change directory to `plugins/woocommerce/tests/performance`
Run all requests with 
`k6 run tests/simple-all-requests.js`

3. Check tests pass in CI

### Other information:




### FOR PR REVIEWER ONLY:

-   [] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.
